### ISSUE TITLE
New user registration: Something isn't quite right yet! Please review the error below

### ISSUE THREAD

**Issue by ralfr** [mastodon/mastodon#21987]
### Steps to reproduce the problem

My instance has been open for registrations and users have registered successfully.
Half an our ago, users start reporting that the web UI returns

`Something isn't quite right yet! Please review the error below`

when submitting the new user form.

I investigated the following:

- No errors in nginx logs.
- `journalctl` doesn't return any errors for `mastodon-web`.
- Postgres is all green.
- All Sidekiq processes are stable and all queues are down to zero.
- No JavaScript errors in the JS Console on the frontend.
- SMTP is working. Other activities requiring SMTP (password resets, etc.) work.
- When switching registration to Waitlist, registration works.

Do you guys have any ideas how to further troubleshoot?

### Expected behaviour

Registrations should work like they always did.

### Actual behaviour

n/a

### Specifications

Mastodon 4.0.2, non-docker

**Comment by ralfr** [comment]
Observation: When switching to Waitlist > Open again, the problem is gone.

**Comment by ClearlyClaire** [comment]
Have any of you managed to reproduce the issue yourself? If so, was any of the registration fields shown to be in error? Do you have server rules set? How do you did you switch to open registrations and back?

**Comment by ClearlyClaire** [comment]
My current guess is that you somehow ended up in a state where everyone is allowed to join without approval but a reason to join is required (thus not accepting an empty reason, but not showing the input either). I'm not completely sure how such a state could be triggered, though, but I'll see what I can do about it. Does your `/admin/settings/registrations` look like that?

![image](https://user-images.githubusercontent.com/384364/206199504-893d902e-7eef-47af-9813-9657cefde07d.png)

**Comment by ralfr** [comment]
Thanks, @ClearlyClaire for looking into this. Unfortunately, I cannot check whether the "Require a reason to join" checkbox was activated even when registrations are completely open.

However, you've probably pointed into a direction I haven't investigated, yet.

Due to a lack of customisability, I've added the custom CSS below to change the text/label for the "reason" form input field. Does this maybe cause an issue when registrations are completely open? I know that there are other admins out there, who are using a similar "hack".

```
div.input.user_invite_request_text label {
  visibility: hidden;
  position: relative;
}
div.input.user_invite_request_text label::before {
  visibility: visible;
  top: 0;
  left: 0;
  content: "Tell us why you want to join home.social! Please remember, we currently support contributions in English and German.";";
}
```

**Comment by ClearlyClaire** [comment]
I don't think this customization should cause any issue.

**Comment by ralfr** [comment]
Today, users reported the error again. I checked and the checkmark at "require a reason" was NOT set. It's super annoying since I lost an entire day of user registrations. Are there any logs we can check to drill down on this issue?

**Comment by ralfr** [comment]
> same problem

Have you been able to resolve the issue. It surfaced here again and it's super annoying. Especially the fact, that no reason is given whatsoever and there are no log entries anywhere.

**Comment by ClearlyClaire** [comment]
No, there wouldn't be logs. Can you ask the user to provide a screenshot? Do your registrations require review? Anything else that comes to mind?

**Comment by ralfr** [comment]
> No, there wouldn't be logs. Can you ask the user to provide a screenshot? Do your registrations require review? Anything else that comes to mind?

I switched back to Manual Approval, since Open Registrations apparently didn't work the entire day. Unfortunately, I haven't noticed until a user dropped me an email. I can switch back to open registrations and see whether the problem is still there. Then we can probably troubleshoot?

**Comment by ClearlyClaire** [comment]
Sounds good! Did you ever check the “require a reason to join”?

**Comment by ralfr** [comment]
> Sounds good! Did you ever check the “require a reason to join”?

Yes. Whenever we switch to manual registrations (aka review) we ask users to provide a reason. We usually switch via `tootctl settings registrations approved --require_reason`. I wonder whether submitting the command and afterwards doing a `tootctl settings registrations open` might end up in a situation where registrations are open but the "require reason" flag might not get cleared accordingly?

**Comment by ralfr** [comment]
This could explain, why switching back and force **through the user interface (!)** resolves the issue.

**Comment by ClearlyClaire** [comment]
Then yes, that appears to be the issue, and it should be fixed by #22127
